lib/entities/validated_entity.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class ValidatedEntity
+  def initialize
+    @errors = []
+  end
+
+  def validate
+    raise NotImplementedError
+  end
+
+  def valid?
+    validate
```

Comment:
each time you use valid? you will call validate

Reply:
yes